src/main/java/com/example/notification/repository/PriceAlertRepository.java

```diff
@@ -14,5 +15,8 @@ public interface PriceAlertRepository extends JpaRepository<PriceAlert, Long> {
     List<PriceAlert> findAll();
 
     @Query("Select p from PriceAlert p where p.user.chatId = :chatId and p.currencyName = :currencyName")
-    Optional<PriceAlert> findByUserIdAndCurrency(@Param("chatId") Long chatId, @Param("currencyName") String currencyName);
+    Optional<PriceAlert> findByUserIdAndCurrency(@Param("chatId") Long chatId, @Param("currencyName") Currency currencyName);
```

Comment:
I tried using your suggestion, but unfortunately, it doesn't find the records. There is an assumption that this might be related to Enums, as Currency is an Enum. In MySQL, this issue was resolved with Hibernate 6.2, but it persists in PostgreSQL